### PR TITLE
SDL1.2: Add SNDIO support

### PR DESCRIPTION
**Pull request by archfan**
About time.

See: 
http://cvsweb.openbsd.org/cgi-bin/cvsweb/ports/devel/sdl/patches 
https://svnweb.freebsd.org/ports/head/audio/deadbeef/

### CODE REVIEW

File: srcpkgs/SDL/template [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+configure_args="--enable-alsa --enable-sndio --disable-esd --disable-rpath
  --enable-clock_gettime --disable-nas --disable-arts --disable-x11-shared
  --disable-alsa-shared --disable-pulseaudio-shared --disable-video-dga
  --disable-osmesa-shared $(vopt_enable aalib video-aalib)
  $(vopt_enable pulseaudio) $(vopt_enable opengl video-opengl)
  $(vopt_if x11 '-enable-video-x11-xrandr --enable-video-x11-vm --enable-video-x11-xv')"
 short_desc="Simple Directmedia Layer"
 maintainer="Juan RP <xtraeme@voidlinux.eu>"
-homepage="http://www.libsdl.org/"
 license="LGPL-2.1"
```
Comment (maxice8) [inline srcpkgs/SDL/template]: please fix license while you're at it

Comment (Oreo639) [reply srcpkgs/SDL/template]: I know this pr is closed, but what maxie means is fix it so it uses the proper spdx identifier. https://spdx.org/licenses/
It needs to be updated to either LGPL-2.1-only or LGPL-2.1-or-later

Comment (archfan) [reply srcpkgs/SDL/template]: Thank you. It's done!

**Comment by archfan** [comment]
I have finally figured out how to squash commits. Git is not that bad after all.

**Comment by archfan** [comment]
Will this get merged?

Asking for a friend.

File: srcpkgs/SDL/patches/SDL-1.2.15-sndio.patch [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
++
++/* Old variable names */
++#define stream			(this->hidden->stream)
++#define parent			(this->hidden->parent)
++#define mixbuf			(this->hidden->mixbuf)
++#define mixlen			(this->hidden->mixlen)
++#define hdl			(this->hidden->hdl)
++
++#endif /* _SDL_sndioaudio_h */
++
```
Comment (Gottox) [inline srcpkgs/SDL/patches/SDL-1.2.15-sndio.patch]: That's a massive patch. is there a chance for upstreaming sndio support?

Comment (archfan) [reply srcpkgs/SDL/patches/SDL-1.2.15-sndio.patch]: SNDIO support has been merged a long time ago but the SDL team never bothered to create a new stable 1.2 release. I could probably trim down this patch a little bit.

**Pull request closed without merging**